src/policy/policy.cpp

```diff
@@ -62,8 +62,10 @@ CAmount GetDustThreshold(const CTxOut& txout, const CFeeRate& dustRelayFeeIn)
     return dustRelayFeeIn.GetFee(nSize);
 }
 
-bool IsDust(const CTxOut& txout, const CFeeRate& dustRelayFeeIn)
+bool IsDust(const CTxOut& txout, const CFeeRate& dustRelayFeeIn, bool permit_dust_anchor)
 {
+    if (permit_dust_anchor && txout.scriptPubKey.IsPayToAnchor()) return false;
```

Comment:
I think this would be better if you just did `if (txout.sPK.IsPayToAnchor()) return 0;` in `GetDustThreshold()`, and moved the `} else if (whichType == ANCHOR && !permit_anchor) {` check to be prior to the dust one? (Then you don't need the extra parameter for `IsDust()`)

I don't think there's much point special casing "non-dust OP_TRUE output" as being okay if you disable ephemeral anchors; old nodes still won't relay it, and future nodes will be doing v3 package relay anyway.

Reply:
Well it's more if someone turns off the feature, they may still think something is not dust based on various interfaces. I'm all for simpler changes of course...

Reply:
It's a signet only feature that's not even targeted at mainnet ("pseudo"), and doesn't have a use case ("i want to send meaningful amounts of funds to whoever mines this tx, but in a more complicated way than dropping it to fees")... Yeah, people could think weird edge cases are okay, but...

Reply:
I'm attempting to minimize the diff of this PR from the one that could go towards Core master, and thinking aloud. I think going with simpler is the right default choice; complexity can be tacked on later.